code/modules/mob/living/simple_animal/hostile/hostile.dm

```diff
@@ -229,6 +234,23 @@ mob/living/simple_animal/hostile/hitby(atom/movable/AM as mob|obj,var/speed = TH
 	target_mob = null
 	return
 
+/mob/living/simple_animal/hostile/proc/Check_fire(target_mob)
+	if(!target_mob)
+		return FALSE
+
+	var/target_hit = FALSE
+	var/friendly_hit = FALSE
+
+	for(var/atom/M in check_trajectory(target_mob, src, pass_flags=PASSTABLE))

```

Comment:
You're doing two `istype` checks here. Just declare the variable of the loop as `var/A`.